Source/Synthesis.Solr/ContentSearchExtensions.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Sitecore.ContentSearch;
+using Sitecore.ContentSearch.Diagnostics;
+using Sitecore.ContentSearch.Linq.Common;
+using Sitecore.ContentSearch.Utilities;
+using Sitecore.Data;
+using Sitecore.Data.Items;
+using Sitecore.Exceptions;
+using Synthesis.ContentSearch.Lucene;
+using Synthesis.Synchronization;
+using Sitecore.ContentSearch.SolrProvider;
+using Synthesis.Solr.ContentSearch;
+using Sitecore.ContentSearch.Linq.Parsing;
+using Sitecore.ContentSearch.Linq.Solr;
+
+namespace Synthesis.Solr
+{
+	public static class ContentSearchExtensions
```

Comment:
Instead of having its own query extensions, we should implement a DI model that allows registering mapper overrides (e.g. at https://github.com/kamsar/Synthesis/blob/master/Source/Synthesis/ContentSearchExtensions.cs#L75).

Implementing this as a sitecore pipeline ('getSynthesisExecutionContexts'?) may make the most sense (e.g. with a IProviderSearchContext in the pipelineargs and another property for executioncontexts that processors may add to. Then the SOLR provider can simply patch in a pipeline processor that knows how to handle SolrSearchContext and override the execution context, without Synthesis core taking a dep on SOLR.